src/averageAges.js

```diff
@@ -55,10 +68,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredWomen = people.filter(person =>
+    person.sex === 'f' && people.some(child => person.name === child.mother)
+  );
+
+  let countOfChildren = 0;
+
+  const sum = filteredWomen.reduce(
+    (total, mother) => {
+      // looking for all the children of the mother
+      const children = people.filter(child => onlyWithSon
+        ? child.mother === mother.name && child.sex === 'm'
+        : child.mother === mother.name
+      );
+
+      // calculate the sum of all age differences for each child
+      return total + children.reduce((totalOfMother, childOfMother) => {
+        countOfChildren++;
+
+        return totalOfMother
+          + childOfMother.born
+          - (people.find(person => person.name === childOfMother.mother).born);
+      }, 0
```

Comment:
```suggestion
      return total + children.reduce((totalOfMother, childOfMother) => {
        return ( totalOfMother
          + childOfMother.born
          - (people.find(person => person.name === childOfMother.mother).born)) / children.length;
      }, 0
```
It will be the same I guess

Reply:
In this function, we don't have a single array to find out the number of operations. In order to know by what number to divide the sum, I simply made a counter instead of creating another array of children